### PR TITLE
Adding a single-product route and view. 

### PR DESCRIPTION
**Pull request by wonmaungthein**
Adding a single-product route and view. This closes issues #9.

### CODE REVIEW

File: controllers/apiController.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const {MongoClient} = require('mongodb');
```
Comment (kabaros) [inline controllers/apiController.js]: why is there an apiController - delete it.

File: helper/view.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const {MongoClient} = require ('mongodb');
+const mongoConnection = process.env.MONGODB_URI || 'mongodb://localhost:27017/shopcentre';
+
+const singleproduct = (query, sucessCallBack) => {
```
Comment (kabaros) [inline helper/view.js]: you don't need a new function, you can just use the function to getProducts in `dbClient` and pass a different query.

File: package.json [hunk 3]
```diff
@@ -12,8 +12,11 @@
     "express": "~4.15.2",
     "hbs": "~4.0.1",
     "mongodb": "^2.2.31",
-    "mongoose": "^4.11.9",
+    "mongoose": "^4.11.10",
     "morgan": "~1.8.1",
     "serve-favicon": "~2.4.2"
+  },
+  "devDependencies": {
+    "eslint": "^4.6.1"
```
Comment (kabaros) [inline package.json]: remove devDependencies. you can run `npm uninstall --save-dev eslint` ... you shouldn't have followed the instructions by Mohamed. These are still in a Pull Request. They haven't been merged into master yet, so they could be wrong (which is the case here)

File: routes/index.js [hunk 4]
```diff
@@ -19,4 +20,27 @@ router.get('/', function (req, res, next) {
   dbClient.getproducts({}, callBack)
 });
 
-module.exports = router;
\ No newline at end of file
+
+router.get('/products/:urlPath', function (req, res, next) {
+
```
Comment (kabaros) [inline routes/index.js]: very good.

File: routes/index.js [hunk 5]
```diff
@@ -19,4 +20,27 @@ router.get('/', function (req, res, next) {
...
+    }
+    else {
+      res.render('single-product', {
+        title: product[0].title,
+        description: `This is more about ${product[0].title}`,
+        product,
+      });
+    }
+  }
+  dbClient.getproducts({ urlPath }, callBack);
```
Comment (kabaros) [inline routes/index.js]: what does this do { urlPath }?

File: package.json (hunk 3 above)
Comment (wonmaungthein) [reply package.json]: I have done this part..

File: helper/view.js (hunk 2 above)
Comment (wonmaungthein) [reply helper/view.js]: I have deleted vies.js file.

File: controllers/apiController.js (hunk 1 above)
Comment (wonmaungthein) [reply controllers/apiController.js]: Have deleted now.

File: routes/index.js (hunk 5 above)
Comment (wonmaungthein) [reply routes/index.js]: I deleted urlPath and tried to run it and it works.. So, I think I don't need it and I have deleted it..

Comment (kabaros) [reply routes/index.js]: it should not work without urlPath. I just wanted to see if you understand how the syntax works

Comment (kabaros) [reply routes/index.js]: so leave urlPath

**Pull request merged**